perma_web/perma/views/common.py

```diff
@@ -206,40 +208,77 @@ def contact(request):
     """
     Our contact form page
     """
+    def affiliation_string():
+        affiliation_string = ''
+        if request.user.is_authenticated():
+            if request.user.registrar:
+                affiliation_string = "{} (Registrar)".format(request.user.registrar.name)
+            else:
+                affiliations = ["{} ({})".format(org.name, org.registrar.name) for org in request.user.organizations.all().order_by('registrar')]
+                if affiliations:
+                    affiliation_string = ', '.join(affiliations)
+        return affiliation_string
+
+    def handle_registrar_fields(form):
+        if getattr(request.user, 'is_organization_user', ''):
+            registrars = set(org.registrar for org in request.user.organizations.all())
+            if len(registrars) > 1:
+                form.fields['registrar'].choices = [(registrar.email, registrar.name) for registrar in registrars]
+            if len(registrars) == 1:
+                form.fields['registrar'].widget = widgets.HiddenInput()
+                email = registrars.pop().email
+                form.fields['registrar'].initial = email
+                form.fields['registrar'].choices = [(email,email)]
+        else:
+            del form.fields['registrar']
+        return form
+
+    def should_send_to_registrar():
```

Comment:
FWIW, for the AnonymousUser trick to work, one has to remove the @cached_property decorated from LinkUser.is_organization_user(). That seems...... potentially like a risk worth taking? Do you have memories of performance problems related to that property?